packages/next/src/shared/lib/router/utils/route-regex.ts

```diff
@@ -83,35 +83,30 @@ export function getRouteRegex(normalizedRoute: string): RouteRegex {
  * number of characters.
  */
 function buildGetSafeRouteKey() {
-  let routeKeyCharCode = 97
-  let routeKeyCharLength = 1
+  let i = 0
 
   return () => {
     let routeKey = ''
-    for (let i = 0; i < routeKeyCharLength; i++) {
-      routeKey += String.fromCharCode(routeKeyCharCode)
-      routeKeyCharCode++
-
-      if (routeKeyCharCode > 122) {
-        routeKeyCharLength++
-        routeKeyCharCode = 97
-      }
+    let j = ++i
+    while (j > 0) {
+      routeKey += String.fromCharCode(97 + ((j - 1) % 26))
+      j = Math.floor((j - 1) / 26)
     }
     return routeKey
   }
 }
 
 function getSafeKeyFromSegment({
+  getSafeRouteKey,
   segment,
   routeKeys,
   keyPrefix,
 }: {
+  getSafeRouteKey: () => string
   segment: string
   routeKeys: Record<string, string>
   keyPrefix?: string
 }) {
-  const getSafeRouteKey = buildGetSafeRouteKey()
```

Comment:
This can't be called inside of this function, but must be hoisted outside to keep state in the getSafeRouteKey closure